web-ui/src/main/resources/catalog/components/history/GnHistoryDirective.js

```diff
@@ -144,6 +144,36 @@
                 });
               };
 
+              scope.restoreHistoryElement = function(statusId) {
+                confirmMessage = $translate.instant('confirmRestore');
+                if($window.confirm(confirmMessage)) {
+                    gnRecordHistoryService.restoreHistoryElement(statusId).then(function(r) {
```

Comment:
I think to get working the UI code with `data-gn-click-and-spin` the promise should be returned otherwise no spinning is displayed.